std-bits/table/src/main/java/org/enso/table/data/column/builder/Builder.java

```diff
@@ -96,4 +100,9 @@ public static Builder getForType(StorageType type, int size) {
 
   /** @return the current storage type of this builder */
   public abstract StorageType getType();
+
+  /** @return any problems that occurred when building the Storage. */
+  public AggregatedProblems getProblems() {
+    return AggregatedProblems.of();
+  }
```

Comment:
I don't think this is a great pattern, in fact I think it is bad.

The proper solution would be to modify `seal()` to return a `WithAggregatedProblems<Storage<?>>` - that would ensure that the problems are handled throughout the codebase.

But we use `seal()` in really many places and I realised ensuring all of this is nicely handled will just make this PR far too large - thus I decided this will be better done as a separate task - #7514

Reply:
In general I think there's too much need to handle problem propagation explicitly in the Enso code, but I don't know enough about the big picture to suggest a solution yet.

Reply:
> In general I think there's too much need to handle problem propagation explicitly in the Enso code, but I don't know enough about the big picture to suggest a solution yet.

I'm not sure if I understand what you mean here? If anything, how error propagation in Enso code works is completely separate of how we do it in Java helpers.

I think the propagation on the Enso side works really well, although indeed there may be some places where we could want to improve them.